### ISSUE TITLE
Does Femah need to be sealed and its public members internal static?

### ISSUE THREAD

**Issue by lholman** [lloydstone/femah#47]
I found myself wanting to verify some expectations on Femah.cs, namely Femah.EnableFeature, Femah.SetSwitchType and Femah.SetFeatureAttributes.  Femah being a sealed class and those being internal static methods pretty much negate me mocking it though.

What's the design decision there?

**Comment by hotstone** [comment]
The design decision was to minimise the amount of noise in the interface we expose to the user of the library.  SetSwitchType, SetFeatureAttributes and EnableFeature are all expected to be invoked by other parts of Femah itself (namely the API and, currently at least, the UI), not by the user of the library.  If the user wishes to set the switch type, set the custom attributes or enable/disable a switch, they're expected to do it by either the RESTful API or the UI.

Also, I was thinking that code like this:

```
Femah.EnableFeature(MyFeatureSwitches.SomeSwitch, true);
if ( Femah.IsFeatureOn(MyFeatureSwitches.SomeSwitch) )
{ 
...
}
```

doesn't really make sense.

I also wanted to enforce the concept that the logic behind switches should be in the SwitchType classes themselves, so was also trying to avoid smartypants people from doing this:

```
// You only get the new checkout OR the new header!
Femah.EnableFeature(MySwitches.NewHeader, !Femah.IsFeatureOn(MySwitches.NewCheckout));
```

Maybe this is breaking SRP though, by indirectly having the Femah class responsible for two things (or, so to speak, two masters)?

**Comment by lholman** [comment]
Okies, I get you, I've kind of forgotten why I wanted to do it now, but it was actually for the API testing.
How do we test those methods then? Just rely on the API testing, but I kind of get stuck in a cricle there where I can't mock them...

Thoughts?

**Comment by lholman** [comment]
@hotstone I remember now, this is what I was trying to do...
https://github.com/lloydstone/femah/blob/master/Femah.Core.Tests/FemahApiTests.cs#L245

Basically, I wanted to assert that the `ApiResponseBuilder` makes calls to some of the `Femah` methods, namely `Femah.EnableFeature(..)` and `Femah.SetSwitchType(..)`, see here: https://github.com/lloydstone/femah/blob/master/Femah.Core/Api/ApiResponseBuilder.cs#L100

Is this a pointless test really?

**Comment by hotstone** [comment]
@lholman  Yeah, this is an interesting one.  I think the test is really along the lines of "If I call ApiResponseBuilder to update a feature switch, is the feature switch updated" so it would be better to keep the internal method calls abstracted away, and just test that the feature switch is updated as expected.

**Comment by lholman** [comment]
@hotstone Okies, cool, I think I was over thinking it.  I'll close this once I've removed those tests I's silenced.
